src/shillelagh/adapters/api/nglsapi.py

```diff
@@ -110,26 +111,28 @@ def get_formatted_xml_string(self, xml_string: str) -> str:
         return '<pre>' + formatted_xml_string + '</pre>'
 
     def set_params(self, bounds) -> dict:
-        params = {'format': 'json'}
-
-        date_time_predicate = bounds.get('date_time')
-        if date_time_predicate:
-            params['from'] = pytz.utc.localize(date_time_predicate.start).strftime('%Y-%m-%dT%H:%M:%SZ')
-            params['to'] = pytz.utc.localize(date_time_predicate.end).strftime('%Y-%m-%dT%H:%M:%SZ')
-        else:
-            # Get latest 60 days of data.
-            params['for'] = '86400m'
-
-        if self.table not in ['agent_activity', 'text_to_911', 'location_discrepancy', 'location_queries_and_results']:
-            interval_predicate = bounds.get('interval')
-            params['interval'] = interval_predicate.value if interval_predicate else 'hour'
-
-        if self.table in ['psap_ring_time', 'psap_queue_time', 'busiest_hour', 'call_summary']:
-            abandoned_predicate = bounds.get('abandoned_tag')
-            params['abandoned'] = abandoned_predicate.value if abandoned_predicate else 'included'
-
-        if self.table in ['busiest_hour', 'call_duration']:
-            call_type_predicate = bounds.get('call_type')
-            if call_type_predicate:
-                params['terms'] = json.dumps({"localCallType": list(self.deserialize_set(call_type_predicate.value))})
-        return params
+        out_params = {'format': 'json'}
+        for (col_name, col_type, params, default) in [
+            (
+                x.get('column_name'),
+                x.get('type'),
+                x.get("predicate", {}).get("params", {}),
+                x.get("predicate", {}).get("default", {})
```

Comment:
I believe you don't need the default parameter for `x.get("predicate", {})` since you are already checking for `if x.get("predicate")`

Reply:
right on that